Always return a value from `_update_or_create_domain_metrics`

nospame opened the pull request:
Otherwise it will TypeError on unpacking `metrics, __` and fail the whole `update_domain_metrics_for_domains` task

## Technical Summary
Just a bugfix for an in-development feature.

## Safety Assurance

### Safety story
We're not swallowing any additional errors here, just returning a tuple of _something_ that can be unpacked. The caller, `update_domain_metrics_for_domains` already accounts for `metrics` potentially being `None`.

### Automated test coverage
No tests for this particular return value.

### QA Plan
No QA planned.

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

nospame committed:
Always return a value from `_update_or_create_domain_metrics`

Otherwise it will TypeError on unpacking `metrics, __` and fail the whole `update_domain_metrics_for_domains` task

orangejenny commented on corehq/apps/data_analytics/tasks.py:
```diff
@@ -151,6 +151,7 @@ def _update_or_create_domain_metrics(domain, all_stats):
         notify_exception(
             None, message='Failed to create or update domain metrics for {domain}: {}'.format(e, domain=domain)
         )
+        return None, False
```
Nitpick: I'd move this out of the except block, it'd be a tiny bit more robust to future updates.

orangejenny replied:
I was imagining the return statement gets moved out of the `try`...although that would presumably mean that it gets moved below the whole try/except, so then you would still want the `return None, False` to be inside the except.

Really, it just visually catches my attention when a function doesn't end with a return statement on the left-most level of indentation as the function body, and it'll catch my attention again if I read/modify/review this code in the future. But then I read the function and see that it will always return something.

nospame replied:
That makes sense. Would it be clearer/more conventional to always assign `metrics, created` and return those two values? Thinking something like this:
```python
    try:
        domain_obj = Domain.get_by_name(domain)
        metrics_dict = domain_metrics(domain_obj, domain_obj['_id'], all_stats)
        metrics, created = DomainMetrics.objects.update_or_create(
            defaults=metrics_dict,
            domain=domain_obj.name,
        )
    except Exception as e:
        notify_exception(
            None, message='Failed to create or update domain metrics for {domain}: {}'.format(e, domain=domain)
        )
        metrics, created = None, False

    return metrics, created
```

orangejenny replied:
I like that! One return value is better than two.

nospame replied:
https://github.com/dimagi/commcare-hq/pull/35606/commits/d8eb9ba3730e01484aa40276eae244733e20121e

nospame committed:
Use a single return from _update_or_create_domain_metrics

Both in the sense of just returning 'metrics', because we never use 'created', and in having one return statement for the function
